### PR TITLE
Add support for 16- and 8-bit registers

### LINKED ISSUE #8: Could do with being able to use 8-bit registers
I think instructions like `set` can only use the 8-bit registers, like `sete al`. I don't think Fisk provides access to these? Could it? I'm trying to reduce branches in https://github.com/kddnewton/regular_expression using `set` and `cmov`.

@kddnewton

### PR DESCRIPTION
**Pull request by chrisseaton**
Should fix #8, but doesn't quite work yet - see comments inline.

### CODE REVIEW

File: lib/fisk.rb [hunk 1]
```diff
@@ -113,6 +113,29 @@ def value
       end
     end
 
+    AH = Register.new "ah", "r8", 0
+    CH = Register.new "ch", "r8", 1
+    DH = Register.new "dh", "r8", 2
+    BH = Register.new "bh", "r8", 3
```
Comment (chrisseaton) [inline lib/fisk.rb:119]: Do you know how we should differentiate the 8-bit high-registers, compared to the low-registers? They're both `r8`.

**Comment by chrisseaton** [comment]
```
  1) Failure:
FiskTest#test_r8h [/Users/chrisseaton/Documents/fisk/test/test_fisk.rb:364]:
--- expected
+++ actual
@@ -1 +1,3 @@
-"ah, 1"
+# encoding: ASCII-8BIT
+#    valid: true
+"al, 1"


  2) Error:
FiskTest#test_r16:
NoMethodError: undefined method `add_prefix' for #<#<Class:0x00007fa6179d8bf8>:0x00007fa6179d8ab8>
Did you mean?  add_rex
    /Users/chrisseaton/Documents/fisk/lib/fisk/instructions/mov.rb:115:in `encode'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:289:in `encode'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:547:in `block in write_to'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:543:in `each'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:543:in `write_to'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:534:in `to_binary'
    /Users/chrisseaton/Documents/fisk/test/test_fisk.rb:377:in `test_r16'
```

**Comment by chrisseaton** [comment]
Improved - now I just get this error in the generated code?

```
  1) Error:
FiskTest#test_r16:
NoMethodError: undefined method `add_prefix' for #<#<Class:0x00007ff4d6055b68>:0x00007ff4d60559d8>
Did you mean?  add_rex
    /Users/chrisseaton/Documents/fisk/lib/fisk/instructions/mov.rb:115:in `encode'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:293:in `encode'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:551:in `block in write_to'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:547:in `each'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:547:in `write_to'
    /Users/chrisseaton/Documents/fisk/lib/fisk.rb:538:in `to_binary'
    /Users/chrisseaton/Documents/fisk/test/test_fisk.rb:377:in `test_r16'
```

I'm also not sure about the addressing mode for registers like `spl` - seems a different mode entirely?

I don't know those immediately myself though - so I'd be happy just to fix this `undefined method 'add_prefix'`

**Comment by tenderlove** [comment]
> I'm also not sure about the addressing mode for registers like `spl` - seems a different mode entirely?

Ya, I'm not sure, I'll have to check the docs. 

> I don't know those immediately myself though - so I'd be happy just to fix this `undefined method 'add_prefix'`

Same. That makes sense.  Please implement it, if you have time, otherwise I can take a stab at it!

**Comment by chrisseaton** [comment]
`spl` etc use `rex`, even though they're not `rex`-y registers.

**Pull request merged**